python/ray/air/_internal/filelock.py

```diff
@@ -14,7 +14,8 @@ class TempFileLock:
 
     def __init__(self, path: str, **kwargs):
         self.path = path
-        self._lock_dir = Path(tempfile.gettempdir()).resolve() / RAY_LOCKFILE_DIR
+        temp_dir = Path(ray._private.utils.get_ray_temp_dir()).resolve()
```

Comment:
> ray.init(runtime_env={"env_vars": {"TMPDIR": "/tmp/abcdef"}}, _temp_dir="/tmp/ray_tmp_dir")

What would the temp directory be in this case?

Reply:
Actually, it needs to be:

```
ray.init(runtime_env={"env_vars": {"RAY_TMPDIR": "/tmp/abcdef"}}, _temp_dir="/tmp/ray_tmp_dir")
```

to work on all platforms. `TMPDIR` only applies to linux for some reason.

In this case, `ray._private.utils.get_ray_temp_dir() == "/tmp/abcdef"`.

Reply:
See here: https://github.com/ray-project/ray/blob/9acf5832246762cc034d37e3926cb31da7c8fde4/python/ray/_private/utils.py#L73

Reply:
Oh, what is the `_temp_dir="/tmp/ray_tmp_dir"` arg for?

Reply:
@jjyao do you know what the difference is between `RAY_TMPDIR` and `_temp_dir`?

Reply:
The difference between `ray._private.utils.get_ray_temp_dir` and `ray.init(_temp_dir)`.

Reply:
oh, I checked the code and I think it's just our code issue: we have these two paths to set and get temp dir and they are not always consistent. I'll create an issue to track this.

Reply:
https://github.com/ray-project/ray/issues/32962